lib/rust/lexer/definition/src/token.rs

```diff
@@ -101,18 +114,90 @@ impl Token {
         Token{shape,length,offset}
     }
 
-    /// Construct a token representing a text literal.
-    pub fn Text(text:impl Str, offset:usize) -> Token {
-        let str    = text.into();
-        let length = str.chars().count();
-        let shape  = Shape::Text(str);
+    /// Construct a token representing a line of text.
+    pub fn TextLine(style:TextStyle, segments:Vec<Token>, offset:usize) -> Token {
+        let length = style.length() + segments.iter().fold(0,|l,r| l + r.source_length());
+        let shape = Shape::TextLine{style,segments};
+        Token{shape,length,offset}
+    }
+
+    /// Construct a token representing an inline block text literal.
+    pub fn TextInlineBlock
+    ( style       : TextStyle
+    , segments    : Vec<Token>
+    , offset      : usize
+    ) -> Token {
+        let segments_length = segments.iter().fold(0,|l,r| l + r.source_length());
+        let length          = style.length() + segments_length;
```

Comment:
this is the same code as in previosu function

Reply:
But this is an artefact of the definitions of the tokens. It would be wrong to unify them as the commonality is only incidental.

Reply:
If this is artefact of the definitions of tokens then maybe this should be token-like utility?

Reply:
Leaving it!